__tests__/pages/curriculum/index.test.js

```diff
@@ -0,0 +1,284 @@
+import dummyAlertData from '../../../__dummy__/alertData'
+
+// Imported to be able to use expect(...).toBeInTheDocument()
+import '@testing-library/jest-dom'
+
+const getAppQueryMock = {
+  request: { query: GET_APP },
+  result: {
+    data: {
+      session: dummySessionData,
+      lessons: [
+        {
+          ...dummyLessonData[0],
+          id: 1
+        },
+        ...dummyLessonData
+      ],
+      alerts: dummyAlertData
+    }
+  }
+}
+
+const getAppQueryMockWithNoModules = {
+  request: { query: GET_APP },
+  result: {
+    data: {
+      session: dummySessionData,
+      lessons: [
+        {
+          ...dummyLessonData[0],
+          modules: [],
+          id: 1
+        }
+      ],
+      alerts: dummyAlertData
+    }
+  }
+}
+
+const fakeExercise = {
+  moduleId: -1,
+  description: 'exercise desc',
+  answer: 'x',
+  explanation: 'because x is the answer'
+}
+
+const addExerciseMutationMock = {
+  request: {
+    query: ADD_EXERCISE,
+    variables: { ...fakeExercise, moduleId: 1 }
+  },
+  result: jest.fn(() => ({
+    data: {
+      addExercise: { ...fakeExercise }
+    }
+  }))
+}
+
+const addExerciseMutationMockSuccess = {
+  request: {
+    query: ADD_EXERCISE,
+    variables: { ...fakeExercise, moduleId: 1 }
+  },
+  result: jest.fn(() => ({
+    data: {
+      addExercise: { ...fakeExercise, moduleId: 1 }
+    }
+  }))
+}
+
+const addExerciseMutationMockError = {
+  request: {
+    query: ADD_EXERCISE,
+    variables: { ...fakeExercise, moduleId: 1 }
+  },
+  error: new Error('Error')
+}
+
+const mocks = [getAppQueryMock, addExerciseMutationMock]
+const mocksWithError = [getAppQueryMock, addExerciseMutationMockError]
+const mocksWithSuccess = [getAppQueryMock, addExerciseMutationMockSuccess]
+const mocksWithNoModules = [
+  getAppQueryMockWithNoModules,
+  addExerciseMutationMock
+]
+
+const useRouter = jest.spyOn(require('next/router'), 'useRouter')
+const useRouterObj = {
+  asPath: 'c0d3.com/curriculum/js1/mentor',
+  query: {
+    lessonSlug: 'js1'
+  },
+  push: jest.fn()
+}
+
+useRouter.mockImplementation(() => useRouterObj)
+
+const fillOutExerciseForms = async () => {
+  const [description, explanation] = screen.getAllByTestId('textbox')
+  const answer = screen.getByTestId('input1')
+
+  // the type event needs to be delayed so the Formik validations finish
+  await userEvent.type(description, fakeExercise.description, { delay: 1 })
+  await userEvent.type(answer, fakeExercise.answer, { delay: 1 })
+  await userEvent.type(explanation, fakeExercise.explanation, { delay: 1 })
+}
+
+describe('Mentor page', () => {
+  it('should not submit when inputs are empty', async () => {
+    expect.assertions(1)
+
+    const { container } = render(
+      <MockedProvider mocks={mocks}>
+        <MentorPage />
+      </MockedProvider>
+    )
+
+    // Helps the data to resolve
+    await act(() => new Promise(res => setTimeout(res, 0)))
+
+    const dropdownBtn = container.querySelector('#dropdown-lesson')
```

Comment:
Why not use `container.getByTestId('dropdown-lesson')` instead?

Reply:
I agree. Updated the code!